### PR TITLE
implement kubectl get/describe for ReplicaSet

### PR DESCRIPTION
**Pull request by mqliang**
Ref #3024

### CODE REVIEW

**Comment by bgrant0607** [comment]
Thanks!

cc @kargakis @janetkuo @madhusudancs

File: pkg/kubectl/describe.go [hunk 1]
```diff
@@ -475,10 +468,6 @@ func describePod(pod *api.Pod, rcs []api.ReplicationController, events *api.Even
 		fmt.Fprintf(out, "Reason:\t%s\n", pod.Status.Reason)
 		fmt.Fprintf(out, "Message:\t%s\n", pod.Status.Message)
 		fmt.Fprintf(out, "IP:\t%s\n", pod.Status.PodIP)
-		var matchingRCs []*api.ReplicationController
```
Comment (bgrant0607) [inline pkg/kubectl/describe.go]: Good catch.

**Comment by bgrant0607** [comment]
cc @kubernetes/kubectl

File: pkg/kubectl/describe.go [hunk 2]
```diff
@@ -87,6 +87,7 @@ func describerMap(c *client.Client) map[unversioned.GroupKind]Describer {
 		api.Kind("Endpoints"):             &EndpointsDescriber{c},
 		api.Kind("ConfigMap"):             &ConfigMapDescriber{c},
 
+		extensions.Kind("ReplicaSet"):              &ReplicaSetDescriber{c},
```
Comment (bgrant0607) [inline pkg/kubectl/describe.go]: @smarterclayton @nikhiljindal DefaultObjectDescriber (below) was added by PR #4880. 

DefaultObjectDescriber only supports a small subset of resource types. I can't see it used anywhere other than the DefaultObjectDescriber unit test. I checked Openshift Origin also.

That PR was about enabling printing of local objects. I don't see how DefaultObjectDescriber is related to that. Can we rip it out?

File: pkg/kubectl/describe.go [hunk 3]
```diff
@@ -899,6 +888,58 @@ func DescribePodTemplate(template *api.PodTemplateSpec) (string, error) {
...
+	events, _ := d.Events(namespace).Search(rs)
+
+	return describeReplicaSet(rs, events, running, waiting, succeeded, failed)
+}
+
+func describeReplicaSet(rs *extensions.ReplicaSet, events *api.EventList, running, waiting, succeeded, failed int) (string, error) {
+	return tabbedString(func(out io.Writer) error {
+		fmt.Fprintf(out, "Name:\t%s\n", rs.Name)
+		fmt.Fprintf(out, "Namespace:\t%s\n", rs.Namespace)
+		if rs.Spec.Template != nil {
```
Comment (bgrant0607) [inline pkg/kubectl/describe.go]: Could we use DescribePodTemplate, here and in describeReplicationController?

**Comment by bgrant0607** [comment]
Thanks. This looks like a faithful translation of the RC code. I had a suggestion and a question, but this is good enough to start with. We can make additional changes in subsequent PRs.

**Comment by bgrant0607** [comment]
LGTM

**Comment by bgrant0607** [comment]
Tests can be added in a subsequent PR, also.

**Comment by bgrant0607** [comment]
See also #20928

File: pkg/api/unversioned/helpers.go [hunk 4]
```diff
@@ -78,3 +78,17 @@ func SetAsLabelSelector(ls labels.Set) *LabelSelector {
 
 	return selector
 }
+
+// FormatLabelSelector convert labelSelector into plain string
+func FormatLabelSelector(labelSelector *LabelSelector) string {
+	var l string
+	if selector, err := LabelSelectorAsSelector(labelSelector); err != nil {
+		l = selector.String()
```
Comment (janetkuo) [inline pkg/api/unversioned/helpers.go]: L86 should exchange with L88

Comment (janetkuo) [inline pkg/api/unversioned/helpers.go]: Filed #20994

**Pull request merged**